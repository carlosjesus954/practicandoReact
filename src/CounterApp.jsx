import PropTypes from "prop-types";

export const EjercicioCounter = ({value}) =>{
    return(
        <>
            <h1>CounterApp</h1>
            <h2>{ value }</h2>
        </>
    )
}

EjercicioCounter.protoTypes = {
    value: PropTypes.number.Required
}