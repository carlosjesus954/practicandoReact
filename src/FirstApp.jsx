// const newMessage= {
//     message: 'Joselu que es???',
//     title: 'Maricon',
//     edad: 99,
//     capacidadMental: 'ninguna'
// };

// const operacionSuma =()=> {
//     return 4 + 10;
// }
import PropTypes from "prop-types";
export const CarlosWapo = ( {
    title, 
    subTitle,
    name
} ) =>{
    // console.log(props)
    
    return (
        <> 
        <h1 data-testid="test-title">{ title }</h1>
        {/* <h1>Cuantos co√±itos hay? { operacionSuma() }</h1> */}
            {/* <h1>Hola { JSON.stringify(newMessage) }</h1> */}
            {/* <h2>{subTitle}</h2> */}
            <h2>{subTitle}</h2>
            <h2>{subTitle}</h2>
            <p>{name}</p>
            
            <p>Ere gey</p>
        </>
    )
}

CarlosWapo.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired
}
CarlosWapo.defaultProps = {
    // title: 'No hay titulo, pero david esta gordito',
    subTitle: 'queh no hay subtiti',
    name:'carlos jesus'
}