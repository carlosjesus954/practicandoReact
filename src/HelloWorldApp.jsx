export function App(){
    return <h1>joselu mama bicho!!</h1>
}
