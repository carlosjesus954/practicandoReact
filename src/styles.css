* {
    margin: 0;
    box-sizing: border-box;
  }
  img {
    object-fit: cover;
    max-width: 100%;
  }
  li {
    list-style: none;
  }
  a {
    text-decoration: none;
  }
  body{
    background-color: rgb(26, 26, 26);
  }
  h1, h2{
    color: white;
    margin-bottom: 2rem;
    text-align: center;
  }
  button{
    display: block;
    padding: 1rem;
    font-size: 1.2rem;
    cursor: pointer;
    margin-left: auto;
    margin-right: auto;
  }