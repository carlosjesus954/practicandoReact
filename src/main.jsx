import React from 'react';
import ReactDOM from 'react-dom/client';

import { EjercicioCounter } from './CounterApp'
// import { CarlosWapo } from './FirstApp'
import './styles.css';

ReactDOM.createRoot( document.querySelector('#root') ).render(
    <React.StrictMode>
        <EjercicioCounter value={0} />
        {/* <CarlosWapo title='Hola, soy gey, david es como una ballena'/> */}
    </React.StrictMode>
)