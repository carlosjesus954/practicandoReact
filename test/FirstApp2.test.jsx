import { render, screen } from "@testing-library/react";
import { CarlosWapo } from "../src/FirstApp";

describe("Pruebas en FirstApp ", () => {
  const title = "Hola, soy goku";
  const subTitle = "queh no hay subtiti";
  test("Debe de hacer match con el snapshot", () => {
    const { container } = render(<CarlosWapo title={title} />);

    expect(container).toMatchSnapshot();
  });
  test('Debe de mostrar el mensaje "Hola, soy goku"', () => {
    render(<CarlosWapo title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
  test("Debe de mostrar el titulo en un h1", () => {
    render(<CarlosWapo title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });
  test("Debe de mostrar el subtitulo enviado por props", () => {
    render(<CarlosWapo title={title} subTitle={subTitle} />);
    expect(screen.getAllByText(subTitle).length).toBe(2);
  });
});
