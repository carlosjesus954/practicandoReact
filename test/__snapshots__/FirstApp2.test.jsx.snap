// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp  Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy goku
  </h1>
  <h2>
    queh no hay subtiti
  </h2>
  <h2>
    queh no hay subtiti
  </h2>
  <p>
    carlos jesus
  </p>
  <p>
    Ere gey
  </p>
</div>
`;
